clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -3549,6 +3549,38 @@ RValue CodeGenFunction::EmitBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
         llvm::MetadataAsValue::get(Ctx, llvm::MDString::get(Ctx, Kind)));
     return RValue::get(Allow);
   }
+  case Builtin::BI__builtin_allow_sanitize_check: {
+    Intrinsic::ID IntrID = Intrinsic::not_intrinsic;
+    StringRef SanitizerName =
+        cast<StringLiteral>(E->getArg(0)->IgnoreParenCasts())->getString();
+
+    if (SanitizerName == "address" || SanitizerName == "kernel-address") {
```

Comment:
Won't this provoke the users to use "address" and "kernel-address" interchangeably?
If we want this, maybe it should be reflected in the documentation?